dali/pipeline/data/tensor_vector.cc

```diff
@@ -195,18 +203,22 @@ void TensorVector<Backend>::set_pinned(bool pinned) {
   }
 }
 
+
 template <typename Backend>
 bool TensorVector<Backend>::is_pinned() const {
-  // TODO(klecki): do we enforce the same pin status between elements, or prohibit mixing APIs?
   if (state_ == State::contiguous) {
     return tl_->is_pinned();
   }
-  if (tensors_.size() > 0) {
-    return tensors_[0]->is_pinned();
+  if (tensors_.size() == 0) {
+    return pinned_;
+  }
+  for (size_t i = 1; i < tensors_.size(); i++) {
+    assert(tensors_[i]->is_pinned() == tensors_[0]->is_pinned());
```

Comment:
Perhaps instead of enforcing uniformity (which may or may not be wanted), you should check that all relevant tensors are pinned.

Reply:
Maybe enforced uniformity better fits the `curr_tensors_size_` approach? I don't know if we envision non-uniform pin status of tensors in batch at all...